src/tactic/local_cache.lean

```diff
@@ -0,0 +1,65 @@
+import tactic.basic
+
+namespace tactic
+namespace local_cache
+
+section internal
+
+variables {α : Type} [reflected α] [has_reflect α]
+
+meta def mk_full_namespace (ns : name) : name := `local_cache ++ ns
+
+meta def get_data_name (ns : name) : tactic name :=
+do o ← tactic.get_options,
+   let opt := mk_full_namespace ns,
+   match o.get_string opt "" with
```

Comment:
I think you should come up with a better option name than `""`. What about `"cache_name"`?

Reply:
I think this argument is the default return value for when `o` doesn't have something called `opt`, which seems completely reasonable.

Reply:
Indeed, `""` is used to detect when the option isn't present, since some argument must be supplied anyway.

Reply:
Oh ok, I see. Maybe `if o.get_string opt "" = "" then _ else _` would be a clearer way to express it